exp/go.mod

```diff
@@ -1,11 +1,10 @@
 module go.uber.org/zap/exp
 
-go 1.19
+go 1.21
```

Comment:
Increasing the minimum required Go version isn't necessary if we're using `//go:build` directives to support exp/'s slog package for older versions.